python/ray/autoscaler/autoscaler.py

```diff
@@ -246,6 +307,19 @@ def __init__(self,
         self.last_update_time = 0.0
         self.update_interval_s = update_interval_s
 
+        # Node launchers
+        self.launch_queue = queue.Queue()
+        self.num_launches_pending = Counter()
```

Comment:
AtomicInteger()?

Reply:
It's not really atomic since it consists of several Python instructions (using a lock for safety), though this is a little pedantic.